wic/iot2050-swu.wks.in

```diff
@@ -10,8 +10,8 @@
 
 include ebg-sysparts.inc
 
-part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "fedcba98-7654-3210-cafe-5e0710000001"
-part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "fedcba98-7654-3210-cafe-5e0710000002"
+part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_A}"
+part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.squashfs" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_B}"
```

Comment:
The `squashfs` and `verity` is also used as the `IMAGE_TYPEDEP` in recipes-core/images/iot2050-image-swu-example.bb, is it possible to define some variables to hold them?

Reply:
I can't follow yet: What exactly should be shared via variables?

Reply:
make `SWU_ROOTFS_TYPE` a global variable and use it here:

```suggestion
part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.${SWU_ROOTFS_TYPE}" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_A}"
part --source rawcopy --sourceparams "file=${IMAGE_FULLNAME}.${SWU_ROOTFS_TYPE}" --align 1024 --fixed-size 4G --uuid "${ABROOTFS_PART_UUID_B}"
```

Then you can combine some of the `iot2050-swu.wks.in` and `iot2050-swu-secure.wks.in`.

Reply:
Nice idea. Unfortunately, variable resolution in wic does not work on include files, and that would be the only way to share those bits between the two wks files.

Reply:
But I just noticed that we can make B smaller by leaving it empty, just like cip-core does now.